src/util/pull-request-title.ts

```diff
@@ -39,10 +39,12 @@ export function generateMatchPattern(pullRequestTitlePattern?: string): RegExp {
     logger.warn("pullRequestTitlePattern miss the part of '${version}'");
   return new RegExp(
     `^${(pullRequestTitlePattern || DEFAULT_PR_TITLE_PATTERN)
-      .replace('${scope}', '(\\((?<branch>[\\w-.]+)\\))?')
+      .replace('[', '\\[')
```

Comment:
perhaps worth pulling in the dep `escape-string-regexp`, we already rely on it as a deep dep, keep the same tests.

Reply:
Note: v5 of that library is the only one with types and our current dependencies use v1 and v4. Additionally, I was having issues with ESM using that library. I added a TODO to handle all regex characters.

Reply:
Note: v5 of that library is the only one with types and our current dependencies use v1 and v4. Additionally, I was having issues with ESM using that library. I added a TODO to handle all regex characters.